Library/Homebrew/hardware.rb

```diff
@@ -41,6 +41,16 @@ def optimization_flags
       end
       alias generic_optimization_flags optimization_flags
 
+      # Only give values where it is an improvement over rust cpu defaults
+      # Rust already defaults to the oldest supported cpu for each target-triple
+      # Including apple-m1 since Rust 1.71 for aarch64-apple-darwin.
```

Comment:
I'd really rather be explicit here rather than bake in assumptions about the behaviour of `rustc`. None of us track the development of `rustc` that carefully, so if they decide to bump the minimum of `core2` before we do, then this method will no longer behave as expected, and that's likely to go unnoticed for some time.

The reverse (i.e. if we decide to bump the minimum of `core2` before Rust does) is also an issue, albeit a smaller one.